packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -528,8 +531,8 @@ class TextSelectionOverlay {
         renderObject.getEndpointsForSelection(_selection);
 
     final Rect editingRegion = Rect.fromPoints(
-      renderObject.localToGlobal(Offset.zero),
-      renderObject.localToGlobal(renderObject.size.bottomRight(Offset.zero)),
+      renderObject.localToGlobal(Offset.zero, ancestor: _overlay.context?.findRenderObject()),
```

Comment:
Ditto on the `?`s on these two lines.  We need to handle the case where these are null if they actually can be null.

Also, this method gets called before you set `_overlay` above. What if `_overlay` is null?

Reply:
We assert that `_overlay` is not null, and it's set in the constructor. Can you please explain how this method gets called before the constructor?
The reason I added `_overlay` is so that we don't do a lookup every time `_buildToolbar` is called, although maybe that's not much of a concern seeing this should not be called every frame.

Finally, on your comment about it being null... if it's null, then the `localToGlobal`'s `ancestor` parameter would be null, which means the global bounds will be used, which I think is OK in that case.

Maybe we just change it so that we call Overlay.of in this method and not save the `_overlay` member?

Reply:
@dnfield Thanks for the review. The PR has been updated to address your comments.